docs/contrib/keycloak.md

```diff
@@ -0,0 +1,148 @@
+# Keycloak Identity Provider for REMS
+
+Using docker, docker-compose, and Keycloak, you can roll your own local IdP for use with REMS.
+
+## Quick-ish Start
+
+1. **Setting up the Docker files**
+    1. Create a `Dockerfile` for your keycloak. To start, it can be as simple as the following:
+    ```
+    FROM quay.io/keycloak/keycloak:11.0.2
+    ```
+    2. Create a `docker-compose.yaml` file for REMS and your Keycloak IdP. Mount your custom REMS config to `/rems/config/config.edn`. Example:
+    ```
+    version: "3.7"
+    services:
+    rems:
+        mem_reservation: 200m
+        mem_limit: 500m
+        ports:
+        - "127.0.0.1:3001:3000"
+        volumes:
+        - ./services/rems/simple-config.edn:/rems/config/config.edn
+        depends_on:
+        - db
+        - keycloak
+    db:
+        image: postgres:9.6
+        environment:
+        POSTGRES_USER: rems
+        POSTGRES_PASSWORD: remspassword
+        mem_reservation: 30m
+        mem_limit: 150m
+        ports:
+        - "127.0.0.1:5432:5432"
+    keycloak:
+        environment:
+        KEYCLOAK_USER: admin
+        KEYCLOAK_PASSWORD: admin
+        ports:
+        - "3002:8080"
```

Comment:
I think the standard image prints it is accessible in port `9990` but instead I got this to work with the `8080:8080`